### ISSUE TITLE
Implement serialization and deserialization of full federation metadata spec

### ISSUE THREAD

**Issue by gislikonrad** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#1517]
Since WS-Trust is being added, wouldn't it make sense to flesh out the [WsFederationMetadataSerializer](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/brentsch/wstrust/src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs)? It reads the SecurityTokenServiceType just to get the keys from it and then it ignores the other role types.

[The spec](http://docs.oasis-open.org/wsfed/federation/v1.2/os/ws-federation-1.2-spec-os.html) specifies multiple role types, but the most common ones are the SecurityTokenServiceType and ApplicationServiceType. Other elements that would be quite good to be able to read/write would be the IDPSSODescriptor and the SPSSODescriptor.

Is this something that could be useful in the wstrust branch?

**Comment by brentschmaltz** [comment]
@gislikonrad yep, our POR includes bumping up our WsFederationMetadataSerializer support.
Our first step was to support WaSignin for asp.net. 
We now have bigger objectives with some additional WsTrust and WCF federation binding support offered by the WCF team.

**Comment by gislikonrad** [comment]
A bit off topic, but...

@brentschmaltz So the WCF team is working on federation bindings for .net standard? That's great news. Do you know if this includes the CreateChannelWithIssuedToken method on the ChannelFactory. That would be quite useful on a WsTrustChannelFactory.

**Comment by brentschmaltz** [comment]
@gislikonrad currently there is no plan for any ChannelFactory type objects, however some users have expressed interest.
Here is the code for the WsFederationBinding : https://github.com/dotnet/wcf/tree/master/src/System.ServiceModel.Federation

**Comment by brentschmaltz** [comment]
@gislikonrad  We are adding WSTrustChannelFactory and WSTrustChannel to https://github.com/dotnet/wcf

Back to your ask, can you narrow your ask to some high value items.
Full Spec has a lot of things most people don't care about.

**Comment by gislikonrad** [comment]
I've already implemented it on my own, so no worries.
